### PR TITLE
Arbitrary compiler options

### PR DESCRIPTION
**Pull request by B14m3m3**
Added a compiler options object that will be passed to the compiler, to enhance the thing in issue #39.

This should allow the user to specific whatever options there are in the cobc as required. I've added the free flag in the `Hello world` example to show the intended change.

As a small side change i also made was adding a callback to the Fs.unlink as it was giving deprecation errors :).  Currently the callback does nothing.

### CODE REVIEW

File: examples/index.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ Cobol(function () {/*
                    DISPLAY "Hello world".
                    PROGRAM-DONE.
                    STOP RUN.
-                   */}, function (err, data) {
+                   */}, {compileargs:{free:true}}, function (err, data) {
```
Comment (IonicaBizau) [inline examples/index.js]: Can you please write this on multiple lines, so it won't be so condensed?

File: lib/compile/index.js [hunk 2]
```diff
@@ -53,9 +53,7 @@ function Compile(input, options, callback) {
                 x: true,
                 _: input
             };
-            if (options.free) {
-                args.free = options.free;
-            }
+            Object.assign(args, options.compileargs);
```
Comment (IonicaBizau) [inline lib/compile/index.js:56]: This looks good!

**Review by IonicaBizau — CHANGES_REQUESTED** [review]
Just a very small change in the example. Overall, it's good! :)

**Comment by B14m3m3** [comment]
Made the change to make the statement a bit more spacious :)

**Comment by B14m3m3** [comment]
@IonicaBizau I'm kinda new to adding things to open-source :) please notify me if another pull request or something has to be done. I should have fixed the example so that it's less condensed

**Review by IonicaBizau — APPROVED** [review]

**Comment by IonicaBizau** [comment]
@B14m3m3 Sorry, I just missed your last commit. :see_no_evil: Do ping me in such cases. Merging right now. :ship:

**Pull request merged**